lib/module.js

```diff
@@ -1,51 +1,65 @@
 const { join, resolve } = require('path')
+const defu = require('defu')
 const { ensureTemplateFile } = require('./utils')
 
 module.exports = async function (moduleOptions) {
+  const { nuxt, addTemplate } = this
   const options = {
     configPath: 'tailwind.config.js',
-    cssPath: join(this.options.dir.assets, 'css', 'tailwind.css'),
+    cssPath: join(nuxt.options.dir.assets, 'css', 'tailwind.css'),
     exposeConfig: false,
-    ...this.options.tailwindcss,
+    ...nuxt.options.tailwindcss,
     ...moduleOptions
   }
 
-  const configPath = this.nuxt.resolver.resolveAlias(options.configPath)
-  const cssPath = this.nuxt.resolver.resolveAlias(options.cssPath)
+  const configPath = nuxt.resolver.resolveAlias(options.configPath)
+  const cssPath = nuxt.resolver.resolveAlias(options.cssPath)
 
   /*
   ** Generates if not exists:
   ** - tailwind.config.js
   ** - @/assets/css/tailwind.css
   */
-  await ensureTemplateFile(this.options.srcDir, 'tailwind.config.js', configPath)
-  const tailwindCSSExists = await ensureTemplateFile(this.options.srcDir, 'tailwind.css', cssPath)
+  await ensureTemplateFile(nuxt.options.srcDir, 'tailwind.config.js', configPath)
+  const tailwindCSSExists = await ensureTemplateFile(nuxt.options.srcDir, 'tailwind.css', cssPath)
 
   // Include CSS file in project css
   if (tailwindCSSExists) {
-    this.options.css.unshift(cssPath)
+    nuxt.options.css.unshift(cssPath)
   }
 
+  // Watch the Tailwind config file to restart the server
+  if (nuxt.options.dev) {
+    nuxt.options.watch.push(configPath)
+  }
   // This hooks is called only for `nuxt dev` and `nuxt build` commands
-  this.nuxt.hook('build:before', () => {
+  nuxt.hook('build:before', async () => {
     // Fix issue with postCSS that needs process.env.NODE_ENV
     /* istanbul ignore if */
-    if (!this.options.dev && !process.env.NODE_ENV) {
+    if (!nuxt.options.dev && !process.env.NODE_ENV) {
       process.env.NODE_ENV = 'production'
     }
     /*
     ** Set PostCSS config
     */
-    const { postcss } = this.options.build
+    const { postcss } = nuxt.options.build
 
     postcss.preset.stage = 1 // see https://tailwindcss.com/docs/using-with-preprocessors#future-css-features
     postcss.plugins = postcss.plugins || {}
 
+    // Get and extend the Tailwind config
+    let tailwindConfig = require(configPath)
```

Comment:
What would be the advantage of require external file over reading from nuxt.options.tailwind? This makes configuration of module inconsistent with others

Reply:
or maybe we can support both configuration sources?

Reply:
Well the module options is different than the Tailwind config (and keeping the standard of using `tailwind.config.js`)